app/controllers/api/plans_base_controller.rb

```diff
@@ -101,7 +101,13 @@ def destroy
 
     if block_given?
       yield
+    elsif @plan.type == 'ApplicationPlan' # Only Application plans are implemented in React right now
```

Comment:
Safe-guard clause easier to read as well:

```ruby
  return yield if block_given?

  unless @plan.type == 'ApplicationPlan'
    ThreeScale::Deprecation.warn "Plans are being migrated to React and this will no longer be used"

    flash[:notice] = 'The plan was deleted'
    redirect_to plans_index_path 
  end

  json = { notice: 'The plan was deleted', id: @plan.id }
  respond_to do |format|
    format.json { render json: json, status: :ok }
  end
```